.github/workflows/foreman.yml

```diff
@@ -53,13 +62,31 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node }}
+      - name: Generate package-lock.json
+        run: npm install --package-lock-only --no-audit
+      - name: Setup Cache
```

Comment:
this gives exactly 0 speedup

Reply:
I also had the impression the cache was really not intended to be used without a package lock file in your repository.

Reply:
Yeah. But I did create a lock, and it did show a cache hit in the logs, and yet the install step is 65 seconds long in cache and no cache cases :woman_shrugging:

Reply:
mhh, with `npm ci` it's now 45s, which is better, I guess?

Reply:
the benchmark, without any caching, was 95s, now it's 45s+35s (for the lock generation) = 80s. Not that much of a win :(

Reply:
Technically we only need NPM packages installed for integration testing. It's tempting to skip it for other tasks.

And I wonder if we could cache the `package-lock.json` file based on the `package.json` file hash.

Reply:
interesting observation: lock + install with caches takes 70 secs (after restoring a "close enough" cache based on package.json alone), while "install w/o locking first" takes 80secs with the same cache. No idea why.

Reply:
one thing that we could speed up (but I am not sure I like) is to cache `node_modules` itself, as the extraction takes quite some time